ocw_data_parser/course_downloader.py

```diff
@@ -54,10 +54,9 @@ def download_courses(self):
                             with open(dest_filename, "wb+") as f:
                                 s3_client.download_fileobj(
                                     self.s3_bucket_name, obj["Key"], f)
-                        courses.pop(courses.index(course_id))
         
-        # if there are still courses in the list, that means they weren't found on s3
-        if len(courses) > 0:
-            print("The following courses were not found in the s3 bucket {}:".format(self.s3_bucket_name))
-            for course_id in courses:
-                print(" - {}".format(course_id))
+        # make sure everything downloaded right
+        downloaded_courses = os.listdir(self.destination_dir)
```

Comment:
I think the original idea of having an array of courses from which courses are removed when they are found in s3 is better than checking the directory for  course folders. I don't think we can assume that the directory that the data is being copied is blank and this will not catch the problem if the course is moved or removed on s3

Reply:
Hm, yea I didn't think of it that way.  Instead, now I'm just making a list of course ID's found while downloading thru the paginator and comparing that to the desired courses at the end.  Ideally I would have liked to just loop through the courses and create filtered paginators for each one, but since the directory structures are inconsistent you can't determine the source bucket key based on just the course ID.  Additional data would have to be added to courses.json and I really like that it's just a simple list of course ID's that can be used in exactly the same format between this and `ocw-to-hugo`.  Either way, this passes the tests and is based on what is actually found while paginating the bucket.